sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusRetryPolicy.cs

```diff
@@ -130,9 +140,10 @@ internal async ValueTask<TResult> RunOperation<T1, TResult>(
             bool logTimeoutRetriesAsVerbose = false)
       {
             var failedAttemptCount = 0;
+            var tryTimeout = CalculateTryTimeout(0);
+            var remainingRetries = CalculateRemainingRetries(failedAttemptCount);
 
-            TimeSpan tryTimeout = CalculateTryTimeout(0);
-            if (IsServerBusy && tryTimeout < ServerBusyBaseSleepTime)
+            if (IsServerBusy && (tryTimeout * remainingRetries) < ServerBusyBaseSleepTime)
```

Comment:
The compiler does not like this calculation because you cannot multiply a TimeSpan and an integer directly.   Convert the TimeSpans to milliseconds and use that as the basis of this calculation.